scipy/special/cephes/hyp2f1.c

```diff
@@ -590,3 +596,32 @@ static double hyp2f1ra(double a, double b, double c, double x,
 
     return f0;
 }
+
+
+/*
+    15.4.2 Abramowitz & Stegun.
+*/
+static double hyp2f1_neg_c_equal_bc(double a, double b, double x)
+{
+    double k;
+    double err;
+    double collector = 1;
+    double sum = 1;
+    double collector_max = 1;
+
+    if (!(fabs(b) < 1e5)) {
+        return NPY_NAN;
+    }
+
+    for (k = 1; k <= -b; k++) {
+        collector *= (a + k - 1)*x/k;
+        collector_max = fmax(fabs(collector), collector_max);
```

Comment:
Is the complicated `fmax` logic really necessary here?

Reply:
Yes. `fmax` avoids branching instructions for all compilers except MSVC 2010 and MSVC 2008 (and I don't really care that Python 2.7 and Python 3.4 on windows will run slower).

Reply:
In C, it looks more complicated (well, if you're not used to functional programming). But in assembly, it's considerably simpler.

Reply:
In that case the `fmax`/`fmin` code should probably go in `special/_c99compat.h` where the other C99 functions are.